1.8.0 blog post (#94)
go-gitea/blog: content/post/release-of-1.8.0.md

internal organizations will only be visible to logged-in users. Private organizations will
only be visible to those who are members.

_Thanks to [**@techknowlogick**](https://github.com/techknowlogick) &amp; [**@DblK**](https://github.com/dblk)_

## Changelog

* SECURITY
  * Prevent remote code execution vulnerability with mirror repo URL settings (#6593) (#6594)
  * Resolve 2FA bypass on API (#6676) (#6674)
  * Prevent the creation of empty sessions for non-logged in users (#6690) (#6677)
* BREAKING
  * Add "ghost" and "notifications" to list of reserved user names. (#6208)
  * Change sqlite DB path default to data directory (#6198)
  * Adds MustChangePassword to user create/edit API (#6193)
  * Disable redirect for i18n (#5910)
  * Releases API paging (#5831)
  * Allow Macaron to be set to log through to gitea.log (#5667)
  * Don't close issues via commits on non-default branch (#5622)
* FEATURE
  * Add regenerate secret feature for oauth2 (#6291)
  * Expose issue stopwatch toggling via API (#5970)
  * Add other session providers (#5963)
  * Pull request conflict files detection (#5951)
  * Integrate OAuth2 Provider (#5378)

>   * Implement "conversation lock" for issue comments (#5073)

Reviewer: This could be moved up with screenshot

Author: done

only be visible to those who are members.

_Thanks to [**@techknowlogick**](https://github.com/techknowlogick) &amp; [**@DblK**](https://github.com/dblk)_

## Changelog

* SECURITY
  * Prevent remote code execution vulnerability with mirror repo URL settings (#6593) (#6594)
  * Resolve 2FA bypass on API (#6676) (#6674)
  * Prevent the creation of empty sessions for non-logged in users (#6690) (#6677)
* BREAKING
  * Add "ghost" and "notifications" to list of reserved user names. (#6208)
  * Change sqlite DB path default to data directory (#6198)
  * Adds MustChangePassword to user create/edit API (#6193)
  * Disable redirect for i18n (#5910)
  * Releases API paging (#5831)
  * Allow Macaron to be set to log through to gitea.log (#5667)
  * Don't close issues via commits on non-default branch (#5622)
* FEATURE
  * Add regenerate secret feature for oauth2 (#6291)
  * Expose issue stopwatch toggling via API (#5970)
  * Add other session providers (#5963)
  * Pull request conflict files detection (#5951)
  * Integrate OAuth2 Provider (#5378)
  * Implement "conversation lock" for issue comments (#5073)

>   * Feature: Archive repos (#5009)

Reviewer 2: Maybe just drop the `Feature: ` here since it's already a bullet point under `Features`.

Author: done

  * Discord Oauth2 support (#4476)
  * Allow to set organization visibility (public, internal, private) (#1763)
  * Added URL mapping for Release attachments like on github.com (#1707)
* ENHANCEMENT
  * Add support for client basic auth for exchanging access tokens (#6293)
  * Add ability to sort issues by due date (#6206) (#6244)
  * Style tweaks to issue selection (#6196)
  * Increase Username and Orgname MaxSize 35 -> 40 (#6178)
  * Coverage profile with multiple packages (#6167)

>   * Split setting.go to multiple files (#6154)

Reviewer 3: ```suggestion
  * Split setting.go to multiple files (#6154) (#6014)
```

  * Allow labels to contain emoji (#6063)
  * Disable git fsck for mirrored repos by default (#6018)
  * Add default time out for git operations (#6015)

>   * Split setting.go as multiple files (#6014)

Reviewer 4: This could be merged with line 80

Reviewer 3: ```suggestion
```

  * Make dashboard navbar and footer full-width (#6013)
  * Add lang specific font stacks for CJK (#6007)
  * Fix header menu misalignment (#6002)
  * Enhance closed PR and Issue status in the list (#6000)
  * Make navbar full width (#5998)
  * Add option to close issues via commit on a non master branch (#5992)
  * Support n as a line highlight prefix (#5987)
  * Search for org repos (#3031) (#5986)
  * Minor UI tweaks (#5980)

>   * Use native golang SSH library but ssh-keygen when enable built-in SSH server to remove dependent on that command lines (#5976)

Reviewer 2: `Use native golang SSH library but ssh-keygen when using the built-in SSH server to remove the dependency on the command line`

Author: done

  * Dashboard tweaks (#5974)
  * Fixes for repo topic editor (#5971)
  * Display the branch name in the commit view (#5950)

>   * handle milestone events for issues and PR (#5947)

Reviewer 2: `Handle` (Capital `H`)

Author: done

  * Switch to more recent build of xgo (#6070)
  * Add autoprefixer to css build (#6029)
  * Update the version of less (#6010)
  * Make log mailer for testing (#5893)
* DOCS
  * Add more tests and docs for issue indexer, add db indexer type for searching from database (#6144)
  * update default value of `--must-change-password` cli flag (#6032)
  * Update and expand information about building Gitea (#6019)
  * Update U2F Section of app.ini.sample (#5994)
  * Update swagger for release API pagination (#5841)
  * Added docs for the tree api (#5834)
* MISC
  * Add single commit API support (#5843)
  * Add missing GET teams endpoints (#5382)
  * Migrate database if app.ini found (#5290)

To see all user-facing changes that went into the release, check out our
[full changelog](https://github.com/go-gitea/gitea/blob/master/CHANGELOG.md#180---2019-04-20).

A shoutout goes to those who reported and/or fixed security issues in this
release:

* [@techknowlogick](https://github.com/techknowlogick) ([#6674](https://github.com/go-gitea/gitea/pulls/6674))
  * With special thanks to [@jonasfranz](https://github.com/jonasfranz) for making Gitea an OAuth2 provider

>    and the many people who reported the 2FA token security loophole.

Reviewer 3: ```suggestion
and to [0x5c](https://github.com/0x5c), Nils Sandmann, [Max Mehl](https://fsfe.org/about/mehl), and Andreas Shimokawa who all independently reported the API 2FA bypass security loophole.
```

* [@muhammedtiftikci](https://github.com/muhammedtiftikci)
* [@pciavald](https://github.com/pciavald)
* [@pbrackin](https://github.com/pbrackin)
* [@pgodwin](https://github.com/pgodwin)
* [@mporrato](https://github.com/mporrato)
* [@richmahn](https://github.com/richmahn)
* [@sapk](https://github.com/sapk)
* [@sebastian-sauer](https://github.com/sebastian-sauer)
* [@segevfiner](https://github.com/segevfiner)
* [@silverwind](https://github.com/silverwind)
* [@saromanov](https://github.com/saromanov)
* [@sd1998](https://github.com/sd1998)
* [@stevegt](https://github.com/stevegt)
* [@techknowlogick](https://github.com/techknowlogick)
* [@tklein23](https://github.com/tklein23)
* [@typeless](https://github.com/typeless)
* [@wewark](https://github.com/wewark)
* [@Whisprin](https://github.com/Whisprin)
* [@xabufr](https://github.com/xabufr)
* [@xdch47](https://github.com/xdch47)
* [@yasuokav](https://github.com/yasuokav)
* [@zeripath](https://github.com/zeripath)

[PRs](https://github.com/go-gitea/gitea/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Amerged+milestone%3A1.8.0)
and [issues](https://github.com/go-gitea/gitea/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+milestone%3A1.8.0)

> merged in 1.7.0.

Reviewer 3: ```suggestion
merged in 1.8.0.
```

![Authorize TestApp Access](https://docs.gitea.io/authorize.png)

_Thanks to [**@jonasfranz**](https://github.com/jonasfranz)_

## Repositories can now be archived [#5009](https://github.com/go-gitea/gitea/pull/5009)

Repositories in Gitea can now be archived. This prevents the creation of new issues,
pull-requests or pushing to the git repository.

![Archived Repository](/demos/5009/1.png)

![Archived Repository List](/demos/5009/2.png)

_Thanks to [**@kolaente**](https://github.com/kolaente)_

## Private and Internal Organizations [#1763](https://github.com/go-gitea/gitea/pull/1763)

Organizations can now be set to be Public, Internal or Private. Repositories created in
internal organizations will only be visible to logged-in users. Private organizations will
only be visible to those who are members.

![Archived Repository List](/demos/1763/1.png)

_Thanks to [**@techknowlogick**](https://github.com/techknowlogick) &amp; [**@DblK**](https://github.com/dblk)_

## Implement "conversation lock" for issue comments [#5073](https://github.com/go-gitea/gitea/pull/5073)

Conversations can now be locked in similar way to Github.

> ![Locked Comments](/demos/5073/1.png)

Reviewer 2: This screenshot could at least be resized if not swapped...

Author: done
